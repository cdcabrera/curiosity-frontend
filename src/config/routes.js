import React from 'react';
import { helpers } from '../common';

const routes = [
  {
    id: 'rhel',
    path: '/rhel',
    pathParameter: ['RHEL'],
    productParameter: ['RHEL'],
    redirect: null,
    isSearchable: true,
    aliases: ['insights'],
    activateOnError: false,
    disabled: helpers.UI_DISABLED,
    default: false,
    component: React.lazy(() => import('../components/productView/productViewRhel'))
  },
  {
    id: 'rhel-arm',
    path: '/rhel-arm',
    pathParameter: ['RHEL for ARM'],
    productParameter: ['RHEL'],
    redirect: null,
    isSearchable: false,
    aliases: [],
    activateOnError: false,
    disabled: helpers.UI_DISABLED,
    default: false,
    component: React.lazy(() => import('../components/productView/productViewRhel'))
  },
  {
    id: 'rhel-ibmpower',
    path: '/rhel-ibmpower',
    pathParameter: ['RHEL for IBM Power'],
    productParameter: ['RHEL'],
    redirect: null,
    isSearchable: false,
    aliases: [],
    activateOnError: false,
    disabled: helpers.UI_DISABLED,
    default: false,
    component: React.lazy(() => import('../components/productView/productViewRhel'))
  },
  {
    id: 'rhel-ibmz',
    path: '/rhel-ibmz',
    pathParameter: ['RHEL for IBM z'],
    productParameter: ['RHEL'],
    redirect: null,
    isSearchable: false,
    aliases: [],
    activateOnError: false,
    disabled: helpers.UI_DISABLED,
    default: false,
    component: React.lazy(() => import('../components/productView/productViewRhel'))
  },
  {
    id: 'rhel-x86',
    path: '/rhel-x86',
    pathParameter: ['RHEL for x86'],
    productParameter: ['RHEL'],
    redirect: null,
    isSearchable: false,
    aliases: [],
    activateOnError: false,
    disabled: helpers.UI_DISABLED,
    default: false,
    component: React.lazy(() => import('../components/productView/productViewRhel'))
  },
  {
    id: 'openshift-container',
    path: '/openshift-container',
    pathParameter: ['OpenShift Container Platform', 'OpenShift-metrics'],
    productParameter: ['OpenShift Container Platform', 'OpenShift-metrics'],
    redirect: null,
    isSearchable: true,
    aliases: [],
    activateOnError: false,
    disabled: helpers.UI_DISABLED,
    default: false,
    component: React.lazy(() => import('../components/productView/productViewOpenShiftContainer'))
  },
  {
    id: 'openshift-dedicated',
    path: '/openshift-dedicated',
    pathParameter: ['OpenShift-dedicated-metrics'],
    productParameter: ['OpenShift-dedicated-metrics'],
    redirect: null,
    isSearchable: true,
    aliases: [],
    activateOnError: false,
    disabled: helpers.UI_DISABLED,
    default: false,
    component: React.lazy(() => import('../components/productView/productViewOpenShiftDedicated'))
  },
  {
    id: 'satellite',
    path: '/satellite',
    pathParameter: ['Satellite'],
    productParameter: ['Satellite'],
    redirect: null,
    isSearchable: false,
    aliases: [],
    activateOnError: false,
    disabled: helpers.UI_DISABLED,
    default: false,
    component: React.lazy(() => import('../components/productView/productViewSatellite'))
  },
  {
    id: 'satellite-capsule',
    path: '/satellite-capsule',
    pathParameter: ['Satellite Capsule'],
    productParameter: ['Satellite'],
    redirect: null,
    isSearchable: false,
    aliases: [],
    activateOnError: false,
    disabled: helpers.UI_DISABLED,
    default: false,
    component: React.lazy(() => import('../components/productView/productViewSatellite'))
  },
  {
    id: 'satellite-server',
    path: '/satellite-server',
    pathParameter: ['Satellite Server'],
    productParameter: ['Satellite'],
    redirect: null,
    isSearchable: false,
    aliases: [],
    activateOnError: false,
    disabled: helpers.UI_DISABLED,
    default: false,
    component: React.lazy(() => import('../components/productView/productViewSatellite'))
  },
  {
    id: 'optin',
    path: '/optin',
    pathParameter: null,
    productParameter: null,
    redirect: null,
    isSearchable: false,
    aliases: [],
    activateOnError: true,
    disabled: helpers.UI_DISABLED,
    default: false,
    component: React.lazy(() => import('../components/optinView/optinView'))
  },
  {
    id: 'missing',
    path: '/',
    pathParameter: null,
    productParameter: null,
    redirect: '/',
    isSearchable: false,
    aliases: [],
    activateOnError: false,
    disabled: helpers.UI_DISABLED,
    default: true,
    component: React.lazy(() => import('../components/productView/productViewMissing'))
  }
];

export { routes as default, routes };
