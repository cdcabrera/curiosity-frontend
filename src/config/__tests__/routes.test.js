import { routes } from '../routes';

describe('Routes configuration', () => {
  it('should return limited generated routes for testing', () => {
    expect(routes.map(({ path }) => path)).toMatchSnapshot('expected paths');
  });
});
