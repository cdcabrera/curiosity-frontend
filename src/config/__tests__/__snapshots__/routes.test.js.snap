// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Routes configuration should return limited generated routes for testing: expected paths 1`] = `
[
  "/openshift-container",
  "/OpenShift Container Platform",
  "/openshift",
  "/openshift-dedicated-metrics",
  "/openshift-dedicated",
  "/rhacs",
  "/rhel",
  "/insights",
  "/rhods",
  "/optin",
  "/",
]
`;
